### ISSUE TITLE
importpath seemed to be ignored

### ISSUE THREAD

**Issue by shariat** [bazelbuild/rules_go#1194]
I have this use case that I want to to use `genrule` as a source of `proto_library`. If I use `genrule` import path will be wrong. I have something like this in `path/proto/lib`, Note that `importpath` has additional `repo`.
```
genrule(
  name = "lib_copy",
  srcs = ["lib.proto.orig"],
  outs = ["lib.proto"],
  cmd = "cp $(<) $(@)",
)

proto_library(
    name = "lib_proto",
    srcs = ["lib.proto"],
)

go_proto_library(
    name = "lib_go_proto",
    importpath = "repo/path/proto/lib",
    proto = ":lib_proto",
)
```
and another proto depends on this in `path/proto/dep`:
```
proto_library(
    name = "dep_proto",
    srcs = ["dep.proto"],
    deps = [
        "//path/proto/lib:lib_proto",
    ],
)

go_proto_library(
    name = "dep_go_proto",
    importpath = "repo/path/proto/dep",
    proto = ":dep_proto",
    deps = [
        "//path/proto/lib:lib_go_proto",
    ],
)
```
`dep_go_proto` failed with 
```
import of path/proto/lib, which is not a direct dependency
```
In generated `dep.pb.go`, `lib` is imported as  `path/proto/lib` instead of `repo/path/proto/lib`. Removing genrule and using a proto file fixes the problem.

**Comment by ianthehat** [comment]
I am surprised because we have a test for exactly this case in [examples/proto/gostyle](/bazelbuild/rules_go/blob/master/examples/proto/gostyle/BUILD.bazel)
If you could work out what is different between your case and that one, or put up some code that I can check out that replicates the failure I would be happy to look into it.

**Comment by shariat** [comment]
one difference is I don't have `go_package`. I thought `importpath` is enough. Second, I am using gogo. I try to create a test case for this.

**Comment by shariat** [comment]
@ianthehat when `go_package` is important?

**Comment by ianthehat** [comment]
I don't think it is any more, I think I worked around all the cases where it used to matter.

**Comment by shariat** [comment]
Well, removing it from gostyle proto breaks the test

**Comment by ianthehat** [comment]
That's irritating, I will look into it as soon as I finish the thing I am doing.

**Comment by ianthehat** [comment]
Seems to be working for me...

**Comment by shariat** [comment]
https://github.com/shariat/rules_go/commit/2479830e0315e6bcb50f79c154f4a4798ccd9332

**Comment by ianthehat** [comment]
Thanks, that helped.
See if this fix does it for you.

**Comment by shariat** [comment]
works, thanks

**Comment by creachadair** [comment]
> I don't think it is any more, I think I worked around all the cases where it used to matter.

I have a question about this comment: Do you mean that it should no longer be necessary to set the `go_package` option? If so: At what release did that become true?

I'm currently having a very painful time migrating our old Go rules to use these new ones, and am mostly stuck on the `go_proto_library` requirement for this setting—we depend on protos from external repos that do not set the option.

**Comment by ianthehat** [comment]
Yes, go_package is no longer needed.
It was mostly fixed in #1102 although there were still some bugs that turned up in external and generated proto files after that.
